core/src/main/java/org/ehcache/config/CacheConfigurationBuilder.java

```diff
@@ -106,16 +114,17 @@ private CacheConfigurationBuilder(final Expiry<? super K, ? super V> expiry, fin
   }
   
   public CacheConfigurationBuilder<K, V> withClassLoader(ClassLoader classLoader) {
-    this.classLoader = classLoader;
-    return this;
+    return new CacheConfigurationBuilder<K, V>(expiry, classLoader, evictionPrioritizer, evictionVeto, resourcePools, serviceConfigurations);
   }
   
   public CacheConfigurationBuilder<K, V> withResourcePools(ResourcePools resourcePools) {
```

Comment:
Yes, it allows users to share ResourcePools instances instead of a mutable builder in case construction of caches happens in different parts of their code.